### ISSUE TITLE
Bad news:  early EOF  "error trying to git clone a repository" Thank you.

### ISSUE THREAD

**Issue by techhubhub** [petersalomonsen/githttpserver#42]
git clone https://github.com/petersalomonsen/githttpserver
cd githttpserver
npm install
npm run start
go to localhost:5000

enter http://localhost:5000/nikitavoloboev/ts
gets

git clone http://localhost:5000/nikitavoloboev/ts
ERROR 12: early EOF
Bad news:
 early EOF
 
 
 
<img width="1205" alt="image" src="https://github.com/petersalomonsen/githttpserver/assets/158099291/9169045f-e89c-4800-b213-069e986a1d56">

**Comment by lemanschik** [comment]
Its because of hos NEAR Protocol usage i do not exactly know why he putted blockchain into this but i guess thats the cors part or a not needed additional feature.

**Comment by petersalomonsen** [comment]
> Its because of hos NEAR Protocol usage i do not exactly know why he putted blockchain into this but i guess thats the cors part or a not needed additional feature.

The smart contract is used for access control to the git server. This project shows how to set up your own git server and control access to it. Instead of managing a user database on your own, a this is managed in a smart contract on NEAR, and so the server can just ask the smart contract if the user has access, and what access that is allowed.

**Comment by lemanschik** [comment]
@petersalomonsen thanks for making that clear. can you also make clear why you only support lib v 1.7 and not newer git versions? was there a blocker? because i need to do it with git 2.45+

**Comment by petersalomonsen** [comment]
> [@petersalomonsen](https://github.com/petersalomonsen) thanks for making that clear. can you also make clear why you only support lib v 1.7 and not newer git versions? was there a blocker? because i need to do it with git 2.45+

No, it's just because I haven't found the time to update yet. Should be possible to update to the latest libgit2 ( have updated several times ).
